unity-renderer/Assets/DCLPlugins/EmoteAnimations/Tests/EmoteAnimationLoaderShould.cs

```diff
@@ -30,28 +30,91 @@ public void SetUp()
         [TearDown]
         public void TearDown() { Object.Destroy(container); }
 
-        [Test]
-        public void ThrowIfNullContainer() { TestUtils.ThrowsAsync<OperationCanceledException>(loader.LoadEmote(null, new WearableItem(), "female")); }
+        [UnityTest]
+        public IEnumerator ThrowIfNullContainer() => UniTask.ToCoroutine(async () =>
+        {
+            try
+            {
+                await TestUtils.ThrowsAsync<OperationCanceledException>(loader.LoadEmote(null, new WearableItem(), "female"));
+            }
+            catch (Exception e)
+            {
+                // We don't 'throw' the exception because it makes the test fail
+                Assert.IsTrue(e.Message.Contains(loader.MISSING_CONTAINER_ERROR));
+            }
+        });
```

Comment:
My fault, I forgot to make the tests `async`.  I think that's the problem and not the throwing of the exception itself (which we are doing in other tests that are working perfectly).

Have you tried this?
```suggestion
        [UnityTest]
        public IEnumerator ThrowIfNullContainer() => UniTask.ToCoroutine(async () => await TestUtils.ThrowsAsync<OperationCanceledException>(loader.LoadEmote(null, new WearableItem(), "female")));
```

Reply:
Yes I already tried doing it that way but the test fails, I just tried again and that test fails with:
```
System.NullReferenceException: Container cannot be null
  at DCL.Emotes.EmoteAnimationLoader+<LoadEmote>d__9.MoveNext () [0x00022] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Assets/DCLPlugins/EmoteAnimations/EmoteAnimationLoader.cs:23 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at Cysharp.Threading.Tasks.UniTask+ExceptionResultSource.GetResult (System.Int16 token) [0x00001] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Library/PackageCache/com.cysharp.unitask@50a67d8f41/Runtime/UniTask.Factory.cs:205 
  at Cysharp.Threading.Tasks.UniTask+Awaiter.GetResult () [0x00015] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Library/PackageCache/com.cysharp.unitask@50a67d8f41/Runtime/UniTask.cs:312 
  at DCL.Helpers.TestUtils+<ThrowsAsync>d__85`1[T].MoveNext () [0x00024] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Assets/Scripts/Tests/TestUtils.cs:1355 
  Expected: <System.OperationCanceledException>
  But was:  <System.NullReferenceException>
---
at DCL.Helpers.TestUtils+<ThrowsAsync>d__85`1[T].MoveNext () [0x000da] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Assets/Scripts/Tests/TestUtils.cs:1370
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0
at Cysharp.Threading.Tasks.UniTask+ExceptionResultSource.GetResult (System.Int16 token) [0x00001] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Library/PackageCache/com.cysharp.unitask@50a67d8f41/Runtime/UniTask.Factory.cs:205
at Cysharp.Threading.Tasks.UniTask+Awaiter.GetResult () [0x00015] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Library/PackageCache/com.cysharp.unitask@50a67d8f41/Runtime/UniTask.cs:312
at DCL.Helpers.TestUtils+<ThrowsAsync>d__84`1[T].MoveNext () [0x00028] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Assets/Scripts/Tests/TestUtils.cs:1349
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0
at Cysharp.Threading.Tasks.UniTask+ExceptionResultSource.GetResult (System.Int16 token) [0x00001] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Library/PackageCache/com.cysharp.unitask@50a67d8f41/Runtime/UniTask.Factory.cs:205
at Cysharp.Threading.Tasks.UniTask+Awaiter.GetResult () [0x00015] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Library/PackageCache/com.cysharp.unitask@50a67d8f41/Runtime/UniTask.cs:312
at DCL.Emotes.EmoteAnimationLoaderShould+<<ThrowIfNullContainer>b__5_0>d.MoveNext () [0x00040] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Assets/DCLPlugins/EmoteAnimations/Tests/EmoteAnimationLoaderShould.cs:47
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0
at Cysharp.Threading.Tasks.UniTask+ExceptionResultSource.GetResult (System.Int16 token) [0x00001] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Library/PackageCache/com.cysharp.unitask@50a67d8f41/Runtime/UniTask.Factory.cs:205
at Cysharp.Threading.Tasks.UniTask+Awaiter.GetResult () [0x00015] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Library/PackageCache/com.cysharp.unitask@50a67d8f41/Runtime/UniTask.cs:312
at Cysharp.Threading.Tasks.UniTaskExtensions+ToCoroutineEnumerator+<RunTask>d__6.MoveNext () [0x00024] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Library/PackageCache/com.cysharp.unitask@50a67d8f41/Runtime/UniTaskExtensions.cs:805
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0
at Cysharp.Threading.Tasks.UniTaskExtensions+ToCoroutineEnumerator.MoveNext () [0x0003a] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Library/PackageCache/com.cysharp.unitask@50a67d8f41/Runtime/UniTaskExtensions.cs:836
at UnityEngine.TestTools.TestEnumerator+<Execute>d__7.MoveNext () [0x0004e] in /Users/pravus/git/unity-renderer-alt/unity-renderer/Library/PackageCache/com.unity.test-framework@1.1.31/UnityEngine.TestRunner/NUnitExtensions/Attributes/TestEnumerator.cs:46
```

The only way I found of having those tests working fine is with the try-catch

Reply:
I updated the expected exceptions using your approach